helpers/config.py

```diff
@@ -1211,48 +1163,44 @@ def __questions_mongo(self):
-            mongo_root_password = self.__config['mongo_root_password']
+            mongo_user_username = self.__dict['mongo_user_username']
+            mongo_user_password = self.__dict['mongo_user_password']
+            mongo_root_username = self.__dict['mongo_root_username']
+            mongo_root_password = self.__dict['mongo_root_password']
 
             CLI.colored_print("MongoDB root's username?",
-                              CLI.COLOR_SUCCESS)
-            self.__config['mongo_root_username'] = CLI.get_response(
+                              CLI.COLOR_QUESTION)
+            self.__dict['mongo_root_username'] = CLI.get_response(
                 r'~^\w+$',
-                self.__config.get('mongo_root_username'),
+                self.__dict['mongo_root_username'],
                 to_lower=False)
 
-            CLI.colored_print("MongoDB root's password?", CLI.COLOR_SUCCESS)
-            self.__config['mongo_root_password'] = CLI.get_response(
+            CLI.colored_print("MongoDB root's password?", CLI.COLOR_QUESTION)
+            self.__dict['mongo_root_password'] = CLI.get_response(
                 r'~^.{8,}$',
-                self.__config.get('mongo_root_password'),
+                self.__dict['mongo_root_password'],
                 to_lower=False,
                 error_msg='Too short. 8 characters minimum.')
 
             CLI.colored_print("MongoDB user's username?",
-                              CLI.COLOR_SUCCESS)
-            self.__config['mongo_user_username'] = CLI.get_response(
+                              CLI.COLOR_QUESTION)
+            self.__dict['mongo_user_username'] = CLI.get_response(
                 r'~^\w+$',
-                self.__config.get('mongo_user_username'),
+                self.__dict['mongo_user_username'],
                 to_lower=False)
 
-            CLI.colored_print("MongoDB user's password?", CLI.COLOR_SUCCESS)
-            self.__config['mongo_user_password'] = CLI.get_response(
+            CLI.colored_print("MongoDB user's password?", CLI.COLOR_QUESTION)
+            self.__dict['mongo_user_password'] = CLI.get_response(
                 r'~^.{8,}$',
-                self.__config.get('mongo_user_password'),
+                self.__dict['mongo_user_password'],
                 to_lower=False,
                 error_msg='Too short. 8 characters minimum.')
 
-            if (self.__config.get('mongo_secured') != Config.TRUE or
-                mongo_user_username != self.__config.get(
-                        'mongo_user_username') or
-                mongo_user_password != self.__config.get(
-                        'mongo_user_password') or
-                mongo_root_username != self.__config.get(
-                        'mongo_root_username') or
-                mongo_root_password != self.__config.get(
-                        'mongo_root_password')) and \
+            if (self.__dict.get('mongo_secured') is not True or
```

Comment:
This case is one of the rare case where we need to keep it as it. 
`mongo_secured` has no default in the config. Because MongoDB was not secured (i.e. no password was set to access the DB). People will old version of `kobo-install` should not have this boolean in their configuration file. 
In that case, `self.__dict['mongo_secured']` would return a `KeyError` exception. The purpose of this is to avoid prompting the message that `MongoDB root's and/or user's usernames have changed!` on upgrade.
`mongo_secured` is added to user's configuration file when the upgrade is done.

Reply:
Ahh good to know 👍 thanks for explaining that.